src/GafferUI/AnimationGadget.cpp

```diff
@@ -532,21 +739,21 @@ void AnimationGadget::removeKeyframes()
 	ScriptNode *scriptNode = IECore::runTimeCast<Animation::CurvePlug>( first )->ancestor<ScriptNode>();
 	UndoScope undoEnabled( scriptNode, UndoScope::Enabled, undoMergeGroup() );
 
-	for( const auto &keyPtr : m_selectedKeys )
+	for( SelectionSet::KeyContainer::iterator it = m_selectedKeys->m_keys.begin(), itEnd = m_selectedKeys->m_keys.end(); it != itEnd; )
 	{
-		Animation::CurvePlug *parent = keyPtr->parent();
+		Animation::CurvePlug* const parent = ( *it )->parent();
 		if( parent )
 		{
-			parent->removeKey( keyPtr );
+			parent->removeKey( *it++ );
 		}
 	}
 
-	m_selectedKeys.clear();
+	m_selectedKeys->clear();
```

Comment:
Trying to get my head round a couple of things here - see if this makes sense :

- If `!parent` then it looks like we're stuck in an infinite loop because we don't increment `it`. But that shouldn't be possible because SelectionSet guarantees to remove keys when they are unparented, so maybe `assert( parent )` is appropriate instead.
- `removeKey()` will mean that `it` is removed from `m_selectedKeys->m_keys` by the signal handling in SelectedKeys. But that's OK because we've incremented it already, correct?
- We'd expect the final `m_selectedKeys->clear()` to be a no-op, because the signal handling in SelectedKeys will have removed each key when it was unparented.